src/backend/common/queries/database_query.py

```diff
@@ -29,3 +30,15 @@ def fetch_async(self) -> TypedFuture[QueryReturn]:
             # Type-hinting the tasklet decorator is hard, but it consumes
             # the generator and returns the overall future
             return query_result  # pyre-ignore
+
+    def fetch_dict(self, version: int) -> dict:
```

Comment:
nit: `typing.Dict` here

Reply:
Also nit - should we have an 
```
class ApiMajorVersion(enum.IntEnum):
  API_V3 = 3
```

so keep this to a know set of major versions? 

and you can then typehint this to `version: ApiMajorVersion`